gobblin-config-management/src/main/java/gobblin/config/configstore/impl/SimpleVersionFinder.java

```diff
@@ -0,0 +1,80 @@
+package gobblin.config.configstore.impl;
+
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import gobblin.config.configstore.VersionFinder;
+
+
+public class SimpleVersionFinder implements VersionFinder<String> {
+
+  //{v1, V2 ... }
+  public static final String VERSION_PATTERN1 = "[Vv]\\d+";
+  public static final Pattern P1 = Pattern.compile(VERSION_PATTERN1);
+
+  // {v1.0, v1.1, V2.1 ...}
+  public static final String VERSION_PATTERN2 = "[Vv]\\d+\\.\\d+";
+  public static final Pattern P2 = Pattern.compile(VERSION_PATTERN2);
+
+  // {v1.0.0, v1.1.1, V2.1.2 ...}
+  public static final String VERSION_PATTERN3 = "[Vv]\\d+\\.\\d+\\.\\d+";
+  public static final Pattern P3 = Pattern.compile(VERSION_PATTERN3);
+
+  public Collection<String> getValidVersions(Collection<String> input) {
+    if (input == null)
+      return null;
+
+    List<String> res = new ArrayList<String>();
+    for (String s : input) {
+      if (P1.matcher(s).matches() || P2.matcher(s).matches() || P3.matcher(s).matches()) {
+        res.add(s);
+      }
+    }
+
+    return res;
+  }
+
+  @Override
+  public String getCurrentVersion(Collection<String> input) {
```

Comment:
I don't think this will work. For a store with stable versions, we need to be able to preserve the versions even if they are rolled back. This means that the latest (greatest) version may be rolled back and cannot be used as current. I think we have to use the _CONFIG_STORE file to store the latest version.